src/main/java/decoder/Decoder.java

```diff
@@ -14,16 +14,19 @@ public static void decode(InputStream dataStream, OutputStream outputStream) {
     // data file and include headers in the proto definitions so we could
     // address issues with large files, stuff not fitting into RAM, etc.
     try {
```

Comment:
so once we update the reader and writer API's, is anything in the try necessary? is the goal to condense this function to only calling a read method and outputing it as json?